components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -1096,6 +1096,30 @@ public String getServiceProviderNameByClientId(String clientId, String clientTyp
 
     }
 
+    /**
+     * @param inboundKey
+     * @param inboundType
+     * @param tenantDomain
+     * @return
+     * @throws IdentityApplicationManagementException
+     */
+    @Override
+    public String getApplicationResourceIDByInboundKey(String inboundKey, String inboundType,
```

Comment:
Do we need a null check here for the three params passed?

Reply:
added